src/shared/common/components/workstack.jsx

```diff
@@ -312,6 +313,22 @@ class WorkstackAllocate extends Component {
                     }
                 }
                 return <td key={row.uuid + column.dataValueKey} className='govuk-table__cell'>{value}</td>;
+            case ColumnRenderer.DATE_RAW: // date, with Red-Amber-White highlighting
+                if (row.deadlineWarning) {
+                    var date = new Date();
+                    if (new Date(row.deadline) < new Date(date.getFullYear(), date.getMonth(), date.getDate())) {
+                        return <td key={row.uuid + column.dataValueKey} className='govuk-table__cell date-warning'>
+                            <span>{value}</span>
+                        </td>;
+                    } else if (new Date(row.deadlineWarning) < date) {
+                        return <td key={row.uuid + column.dataValueKey} className='govuk-table__cell date-advisory'>
+                            <span>{value}</span>
+                        </td>;
+                    }
+                }
+                return <td key={row.uuid + column.dataValueKey} className='govuk-table__cell date-nowarning'>
+                    <span>{value}</span>
+                </td>;
```

Comment:
Could be refactored to avoid duplication like below. Note I've not tested this refactor:

Also note I have changed the `var` on line 318 to `const date = `
```suggestion
            case ColumnRenderer.DATE_RAW: { // date, with Red-Amber-White highlighting
                let rowClasses = 'govuk-table__cell date-nowarning';
                if (row.deadlineWarning) {
                    const date = new Date();
                    if (new Date(row.deadline) < new Date(date.getFullYear(), date.getMonth(), date.getDate())) {
                        rowClasses = 'govuk-table__cell date-warning';
                    } else if (new Date(row.deadlineWarning) < date) {
                        rowClasses = 'govuk-table__cell date-advisory';
                    }
                }
                return <td key={row.uuid + column.dataValueKey} className={rowClasses}>
                    <span>{value}</span>
                </td>;
            }
```